packages/utils/src/lib/log-results.ts

```diff
@@ -0,0 +1,53 @@
+type PromiseFulfilledResult<T> = {
+  status: 'fulfilled';
+  value: T;
+};
+
+export function logMultipleResults<T>(
+  results: PromiseSettledResult<T>[],
+  messagePrefix: string,
+  succeededCallback?: (result: PromiseFulfilledResult<T>) => void,
+  failedCallback?: (result: PromiseRejectedResult) => void,
+) {
+  if (succeededCallback) {
+    const succeededResults = results.filter(
+      (result): result is PromiseFulfilledResult<T> =>
+        result.status === 'fulfilled',
+    );
+
+    logPluginExecution(
+      succeededResults,
+      `${messagePrefix} successfully: `,
+      succeededCallback,
+    );
+  }
+
+  if (failedCallback) {
+    const failedResults = results.filter(
+      (result): result is PromiseRejectedResult => result.status === 'rejected',
+    );
+
+    logPluginExecution(
+      failedResults,
+      `${messagePrefix} failed: `,
+      failedCallback,
+    );
+  }
+}
+
+export function logPluginExecution<T>(
+  results: (PromiseFulfilledResult<T> | PromiseRejectedResult)[],
+  logMessage: string,
+  callback:
+    | ((result: PromiseFulfilledResult<T>) => void)
+    | ((result: PromiseRejectedResult) => void),
+): void {
+  if (results.length) {
+    console.log(logMessage);
+    results.forEach(
+      callback as (
+        result: PromiseFulfilledResult<T> | PromiseRejectedResult,
+      ) => void,
+    );
+  }
+}
```

Comment:
You can take advantage of the fact that you've already narrowed the types of `succeededResults` and `failedResults` when you call this function. The function parameter types are a little too loose, but you can tighten them up by making the result type a generic type parameter (it'll be narrowed to just one option when you call it), so you then don't need to cast the `callback` in `forEach`.

```suggestion
export function logPluginExecution<
  T extends PromiseFulfilledResult<unknown> | PromiseRejectedResult,
>(results: T[], logMessage: string, callback: (result: T) => void): void {
  if (results.length) {
    console.log(logMessage);
    results.forEach(callback);
  }
}
```

Reply:
Great point, fixed